ceilometer/network/services/port_metering.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2016 Eayun, Inc.
+# All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+from oslo.utils import timeutils
+
+from ceilometer import plugin
+from ceilometer import sample
+from ceilometer import utils
+from ceilometer.openstack.common import log
+from ceilometer.openstack.common import processutils
+
+LOG = log.getLogger(__name__)
+
+IPT_CHAIN_NAME_LEN = 28
+
+
+class _ESPortMeteringPollster(plugin.PollsterBase):
+
+    @staticmethod
+    def _get_ipt_chain_counters(chain):
+        stats = {'packets': 0, 'bytes': 0}
+        cmd = ['iptables', '-t', 'filter', '-L', chain,
+               '-n', '-v', '-x', '-Z']
+        try:
+            (out, _) = utils.execute(*cmd, run_as_root=True)
+        except processutils.ProcessExecutionError as err:
+            LOG.error('Failed to get counters for iptables chain %(chain)s: '
+                      '%(error)s', {'chain': chain, 'error': err})
+            return stats
+
+        lines = out.split('\n')
+        for line in lines[2:]:
+            if not line:
+                break
+            data = line.split()
+            if (len(data) < 2 or
+                not data[0].isdigit() or
+                not data[1].isdigit()
+            ):
```

Comment:
ceilometer/network/services/port_metering.py:52:13: E124 closing bracket does not match visual indentation